### ISSUE TITLE
There's a problem with the code. "optionEqualsValue: true" should be removed.

### ISSUE THREAD

**Issue by nallown** [75lb/handbrake-js#25]
I'm trying to encode a MKV file into a MP4 using Handbrake.js but for some reasons the subtitles aren't getting added to the final encoded video.

I think its a problem with your code because when I remove `optionEqualsValue: true` from your code it works perfectly fine and the subtitles are getting added but when the `optionEqualsValue: true` is in the libraries code then it doesn't work.

This is where I removed that `optionEqualsValue: true` :
https://github.com/75lb/handbrake-js/blob/80dd2a03ead3d2ba580be578242690d2c36ea0f6/lib/Handbrake.js#L79

I'm using this config for encoding the video:
```
{ input: 'nogame.mkv',
  output: 'nogame.mp4',
  encoder: 'x264',
  'encoder-preset': 'medium',
  'encoder-tune': 'animation',
  quality: '20',
  optimize: true,
  audio: 2,
  aencoder: 'faac',
  subtitle: 2,
  'subtitle-burned': 2 }
```

And I tested it with this mkv file: https://www.dropbox.com/s/pfq5ek03jd5thd9/nogame.mkv?dl=0

I've been pulling my hairs out for the past 2 days trying to figure out why the subtitles aren't getting added to the videos, its literally been making me go crazy. When I was encoding the video directly with the shell command (HandBrakeCLI) its been working fine but when I used the HandBrakeJS moddule it just wouldn't work.

Please be my saviour and fix this issue for me.

Can you please remove `optionEqualsValue: true` from your code because that for some reason is breaking the encoder.

**Comment by 75lb** [comment]
Hi, thanks for the detailed issue report. I have reproduced the bug - it is related to #16.

I will let you know once it is fixed.

**Comment by nallown** [comment]
Thank you.

I forgot to mention that the subtitles get added fine if you encode the video using the attributes `--input nogame.mkv --output nogame.mp4 --encoder x264 --encoder-preset medium --encoder-tune animation --quality 20 --optimize --audio 2 --aencoder faac --subtitle 2 --subtitle-burned 2` with the command HandbrakeCLI.

So basically when you add spaces instead of equals signs then it works fine, like it says in #16.

It's mainly the option `--subtitle-burned 2` that needs to have a space with the attribute instead of a equals sign, so it doesn't encode with the subtitles if you add the equals sign there.

**Comment by nallown** [comment]
@75lb do you have an ETA for this? I do a lot of encoding for my website and my team of content managers are waiting for this bug to be fixed.

If you can tell me how you were planning on fixing this then I can try writing the patch for it. Whatever makes you comfortable I guess.

**Comment by 75lb** [comment]
No ETA yet. In the meantime, I suggest you create a fork of the module containing your fixes. 

The fix for this issue (removing blanket use of `--option=value` arg notation) introduces considerable risk of breaking other handbrake features which *require* `--option=value` notation. The fix requires vigilance and plenty of testing which I'll work on. But sorry, I can't give you a ETA yet.

**Comment by nallown** [comment]
no problem, I'll fork it for now.

**Comment by nallown** [comment]
I will check now, give me a little bit of time to test it out

**Comment by nallown** [comment]
Yup, its still a problem. It happens when the `--subtitle-burned` option uses a a equal sign instead of a space.

So instead of having `--subtitle-burned=2` (note the equals sign) it should be `--subtitle-burned 2` (note the space).

**Comment by 75lb** [comment]
Fixed and released in v4.0.3.

I've [added exclusions](https://github.com/75lb/handbrake-js/blob/master/lib/Handbrake.js#L78) for `--preset-import-file`, `--preset-import-gui` and `--subtitle-burned` - these options will no longer be passed to the underlying `HandbrakeCLI` tool using `--options=value` notation. 

I've also improved the error handling to prevent handbrake-js failing silently if this issue arises again, instead you'll get an error message like this: 

```
Other: HandbrakeCLI reported "unknown option", please report this issue: https://github.com/75lb/handbrake-js/issues/new
    at Handbrake._emitOutput (/Users/lloyd/Documents/75lb/handbrake-js/lib/Handbrake.js:217:19)
    at Socket.emit (events.js:210:5)
    at addChunk (_stream_readable.js:308:12)
    at readableAddChunk (_stream_readable.js:289:11)
    at Socket.Readable.push (_stream_readable.js:223:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:182:23) {
  name: 'Other',
  message: 'HandbrakeCLI reported "unknown option", please report this issue: https://github.com/75lb/handbrake-js/issues/new',
  output: '[22:41:58] hb_init: starting libhb thread\n' +
    '[22:41:58] thread 70000578c000 started ("libhb")\n' +
    'unknown option (--preset-import-gui=./tmp/issue-37.json)\n',
  options: {
    'preset-import-gui': './tmp/issue-37.json',
    preset: 'sameWidthSameFps',
    input: './tmp/test.mp4',
    output: './tmp/test.mkv'
  }
}
```
